### PR TITLE
US136314 - lit 2 upgrade breaking

### PR DESCRIPTION
**Pull request by gordon-plumb**
This PR makes the breaking changes required to update to Lit 2 (`lit-html` 2 and `lit-element` 3). The Lit 2 upgrade will be a coordinated effort, so **this PR should not merge at this time**.

Breaking change checklist: 
- [x] Update `lit-html` -> `^2`, `lit-element` -> `^3`, `open-wc/testing` -> `^3` (if used)
- [x] `createRenderRoot` usages -> move from LitElement to ReactiveElement 
- [x] Remove Dependabot Lit rules
- [x] Bump any dependencies to their Lit 2 versions
- [x] Remove any manual lifecycle calls to Lit 1 reactive controllers
- [x] Backwards-compatible changes merged: https://github.com/Brightspace/content-components/pull/213

Renamed API's:
- [x] `UpdatingElement` -> `ReactiveElement`
- [x] `@internalProperty` -> `@state` (we don't use TypeScript decorators)
- [x] `static getStyles()` -> `static finalizeStyles(styles)`
- [x] `_getUpdateComplete()` -> `getUpdateComplete()`
- [x] `NodePart` -> `ChildPart`

Testing checklist: 
- [x] CI/ unit tests pass
- [x] Local testing on demo pages
- [x] Testing component in the LMS

For more information on the Lit 2 upgrade, visit https://lit.dev/docs/releases/upgrade/ or contact team Gaudi.

### CODE REVIEW

File: core/d2l-content-viewer/d2l-content-viewer.js [hunk 1]
```diff
@@ -119,9 +119,9 @@ class ContentViewer extends InternalLocalizeMixin(LitElement) {
 			<d2l-labs-media-player
 				crossorigin="anonymous"
 				media-type="${this._revision.type === ContentType.Video ? 'video' : 'audio'}"
-				metadata=${ifDefined(this._metadata ? this._metadata : undefined)}
```
Comment (gordon-plumb) [inline core/d2l-content-viewer/d2l-content-viewer.js:122]: `ifDefined` directive in an attribute expression now removes the attribute for both `null` and `undefined`

File: package.json [hunk 2]
```diff
@@ -40,9 +40,9 @@
     "@adobe/lit-mobx": "0.0.4",
```
Comment (annabelsegalld2l) [inline package.json:40]: @gordon-plumb Would you be able to bump this to `^2`? lit-mobx released a new version that is compatible with Lit-element 3. It doesn't have any breaking changes, but it changes it's underlying dependency to Lit 2 to match our changes. Here's the diff in the lit-mobx package: https://github.com/adobe/lit-mobx/pull/93/files?diff=split&w=1.

File: package.json [hunk 3]
```diff
@@ -40,9 +40,9 @@
...
     "@brightspace-ui-labs/accordion": "^2",
```
Comment (annabelsegalld2l) [inline package.json:41]: You will also need to bump accordion, file-uploader, and d2l-polymer-behaviors. They are polymer repos that also import Lit components.

File: package.json (hunk 2 above)
Comment (gordon-plumb) [reply package.json:40]: Will do, thanks for the heads-up.

**Review by MichaelKovalchuk — APPROVED** [review]

**Review by MichaelKovalchuk — APPROVED** [review]

**Comment by annabelsegalld2l** [comment]
These changes have been merged as part of https://github.com/Brightspace/content-components/pull/261, so we can close this PR 🎉

**Pull request closed without merging**